worlds/shivers/Rules.py

```diff
@@ -8,58 +8,58 @@
-        and state.has_all({"Cloth Pot Bottom", "Cloth Pot Top", "Cloth Pot Bottom DUPE", "Cloth Pot Top DUPE"}, player)
+    return state.has_all({"Cloth Pot Bottom", "Cloth Pot Top", "Cloth Pot Bottom DUPE", "Cloth Pot Top DUPE"}, player) or \
+        state.has_all({"Cloth Pot Complete", "Cloth Pot Complete DUPE"}, player)
 
 
 def wood_capturable(state: CollectionState, player: int) -> bool:
-    return (state.can_reach("Workshop", "Region", player) or state.can_reach("Blue Maze", "Region", player) or state.can_reach("Gods Room", "Region", player) or state.can_reach("Anansi", "Region", player)) \
-        and state.has_all({"Wood Pot Bottom", "Wood Pot Top", "Wood Pot Bottom DUPE", "Wood Pot Top DUPE"}, player)
+    return state.has_all({"Wood Pot Bottom", "Wood Pot Top", "Wood Pot Bottom DUPE", "Wood Pot Top DUPE"}, player) or \
+        state.has_all({"Wood Pot Complete", "Wood Pot Complete DUPE"}, player)
 
 
 def crystal_capturable(state: CollectionState, player: int) -> bool:
-    return (state.can_reach("Lobby", "Region", player) or state.can_reach("Ocean", "Region", player)) \
-        and state.has_all({"Crystal Pot Bottom", "Crystal Pot Top", "Crystal Pot Bottom DUPE", "Crystal Pot Top DUPE"}, player)
+    return state.has_all({"Crystal Pot Bottom", "Crystal Pot Top", "Crystal Pot Bottom DUPE", "Crystal Pot Top DUPE"}, player) or \
+        state.has_all({"Crystal Pot Complete", "Crystal Pot Complete DUPE"}, player)
 
 
 def sand_capturable(state: CollectionState, player: int) -> bool:
-    return (state.can_reach("Greenhouse", "Region", player) or state.can_reach("Ocean", "Region", player)) \
-        and state.has_all({"Sand Pot Bottom", "Sand Pot Top", "Sand Pot Bottom DUPE", "Sand Pot Top DUPE"}, player)
+    return state.has_all({"Sand Pot Bottom", "Sand Pot Top", "Sand Pot Bottom DUPE", "Sand Pot Top DUPE"}, player) or \
+        state.has_all({"Sand Pot Complete", "Sand Pot Complete DUPE"}, player)
 
 
 def metal_capturable(state: CollectionState, player: int) -> bool:
-    return (state.can_reach("Projector Room", "Region", player) or state.can_reach("Prehistoric", "Region", player) or state.can_reach("Bedroom", "Region", player)) \
-        and state.has_all({"Metal Pot Bottom", "Metal Pot Top", "Metal Pot Bottom DUPE", "Metal Pot Top DUPE"}, player)
+    return state.has_all({"Metal Pot Bottom", "Metal Pot Top", "Metal Pot Bottom DUPE", "Metal Pot Top DUPE"}, player) or \
+        state.has_all({"Metal Pot Complete", "Metal Pot Complete DUPE"}, player)
 
 
 def lightning_capturable(state: CollectionState, player: int) -> bool:
-    return (first_nine_ixupi_capturable or state.multiworld.early_lightning[player].value) \
-        and state.can_reach("Generator", "Region", player) \
-        and state.has_all({"Lightning Pot Bottom", "Lightning Pot Top", "Lightning Pot Bottom DUPE", "Lightning Pot Top DUPE"}, player)
+    return (first_nine_ixupi_capturable or state.multiworld.worlds[player].options.early_lightning.value) \
```

Comment:
I have two questions:

1. Isn't `first_nine_ixupi_capturable` a _function_? If so, don't you need to call it? `first_nine_ixupi_capturable(state, player)`. I think right now, this always evaluates as True (kind of) because functions are truthy

2. I'm not a big fan of getting options from `state.multiworld.worlds[player].options`. I would consider, for functions that need options, passing `state: CollectionState, world: ShiversWorld`. Then you can get `player` from `world.player` and `options` from `world.options`. Even more ideally, you would set a different access rule per option value, that would be faster. But I understand that's a bigger structural change.

Reply:
Quick explanation on that last thing:

```
location.access_rule = lambda state: state.has("Item1", player) and (world.options.foo or state.has("Item 2", player))
```

would be faster if you conditionally set a different rule instead:

```
if world.options.foo:
    location.access_rule = lambda state: state.has("Item1", player)
else:
    location.access_rule = lambda state: state.has_all(["Item1", "Item2"], player)
```

But yeah, it's not a huge deal. The other two things I said are more important

Reply:
1. Fixed

2. Not yet implemented.